### PR TITLE
Prerelease workflow ADR

### PR DESCRIPTION
**Pull request by micgro42**
This PR introduces the Pre-release workflow through an ADR, to make the decision making as transparent as possible. The change includes the ADR itself and any necessary changes to our files to enable the consequences stated in the ADR.

Bug: [T267094](https://phabricator.wikimedia.org/T267094)

### CODE REVIEW

**Comment by itamargiv** [comment]
Oh thank you! :pray:

File: docs/src/00-doc/04_release_management.stories.mdx [hunk 1]
```diff
@@ -24,6 +24,26 @@ Design System changes are supposed to influence application styles, therefore a
 Deciding if a change to the Design System is "breaking" is the responsibility of the developer(s) making the change.  
 It is also their responsibility to inform other relevant roles like UX and PM (if the change is being made at their request) of that, so the additional effort of updating all consuming applications can be taken into consideration.
 
+## Creating a pre-release 
+
+If you want to be able to test components and features prior to their official release, you may create a pre-release. To make a pre-release you need to take the following steps:
+ - To release a new version run the command `npm run bump-version -- [premajor | preminor | prepatch | prerelease]`. This will take care of updating the version number in all package.json files.
+ - Commit and push the changes you made to package files and open a PR.
+ - After the PR is merged create and push a new tag, named after the version number with a "v" prefix and an `alpha` suffix, e.g. `v1.2.3-alpha.0` (0 stands for the first pre-release of this version, for each consecutive pre-release this number must be incremented).
```
Comment (SaiSan-WMDE) [inline docs/src/00-doc/04_release_management.stories.mdx:32]:
```suggestion
 - After the PR is merged, create and push a new tag named after the version number with a "v" prefix and an `alpha` suffix, e.g. `v1.2.3-alpha.0` (0 stands for the first pre-release of this version, for each consecutive pre-release this number must be incremented).
```

Comment (SaiSan-WMDE) [inline docs/src/00-doc/04_release_management.stories.mdx:32]:
```suggestion
 - After the PR is merged, create and push a new tag named after the version number with a "v" prefix and an `alpha` suffix, e.g. `v1.2.3-alpha.0` (0 stands for the first pre-release of this version, for each consecutive pre-release this number must be incremented).
```

Comment (SaiSan-WMDE) [reply docs/src/00-doc/04_release_management.stories.mdx:32]: All I could contribute with 😅  It's all clearly and nicely explained.

**Review by tzhelyazkova — APPROVED** [review]
I generally like the idea :+1: The bigger the DS gets the harder it will be to make stable releases in a timely manner, so doing a pre-release is definitely a good solution.
I really hope this won't be abused to quickly make a component that one needs, pre-release it and then figure out how to make it design system-y.  Ultimately, breaking changes (e.g. deprecating a component, a prop, or changing names) should be very rare.

**Pull request merged**